### PR TITLE
[FLINK-25537][JUnit5 Migration] Module: flink-core with,Package: Configuration

### PR DESCRIPTION
**Pull request by GOODBOY008**
Changes:

- Module: flink-core with,Package: Configuration migrate to junit5

### CODE REVIEW

**Comment by GOODBOY008** [comment]
@Jiabao-Sun PTAL

**Comment by GOODBOY008** [comment]
@Jiabao-Sun @1996fanrui PTAL

File: flink-core/src/test/java/org/apache/flink/configuration/FilesystemSchemeConfigTest.java [hunk 1]
```diff
@@ -21,83 +21,79 @@
...
-        assertNull(justPath.getScheme());
+        URI justPath = new URI(File.createTempFile("junit", null, tempFolder).toURI().getPath());
+        assertThat(justPath.getScheme()).isNull();
 
         try {
             FileSystem.get(justPath);
             fail("should have failed with an exception");
         } catch (UnsupportedFileSystemSchemeException e) {
-            assertTrue(e.getMessage().contains("otherFS"));
+            assertThat(e.getMessage()).contains("otherFS");
```
Comment (1996fanrui) [inline flink-core/src/test/java/org/apache/flink/configuration/FilesystemSchemeConfigTest.java:83]: It's better to use `assertThatThrownBy` instead.

File: flink-core/src/test/java/org/apache/flink/configuration/MemorySizeTest.java [hunk 2]
```diff
@@ -20,243 +20,232 @@
...
-        assertEquals(1L, MemorySize.parse("1 K").getKibiBytes());
-        assertEquals(1L, MemorySize.parse("1 M").getMebiBytes());
-        assertEquals(1L, MemorySize.parse("1 G").getGibiBytes());
-        assertEquals(1L, MemorySize.parse("1 T").getTebiBytes());
+    void testUpperCase() {
+        assertThat(MemorySize.parse("1 B").getBytes()).isEqualTo(1L);
+        assertThat(MemorySize.parse("1 K").getKibiBytes()).isEqualTo(1L);
+        assertThat(MemorySize.parse("1 M").getMebiBytes()).isEqualTo(1L);
+        assertThat(MemorySize.parse("1 G").getGibiBytes()).isEqualTo(1L);
+        assertThat(MemorySize.parse("1 T").getTebiBytes()).isEqualTo(1L);
```
Comment (1996fanrui) [inline flink-core/src/test/java/org/apache/flink/configuration/MemorySizeTest.java:147]: isOne

**Review by 1996fanrui — APPROVED** [review]
Thanks @GOODBOY008 for the contribution!

Overall is fine, I only left 2 minor comments, please take a look in your free time, thanks~

File: flink-core/src/test/java/org/apache/flink/configuration/ConfigurationConversionsTest.java [hunk 3]
```diff
@@ -52,7 +51,9 @@ public class ConfigurationConversionsTest {
 
     private Configuration pc;
 
-    @Before
+    @Parameter private TestSpec testSpec;
+
+    @BeforeEach
     public void init() {
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/configuration/ConfigurationConversionsTest.java:57]: The visibility of class and methods can be package default.

File: flink-core/src/test/java/org/apache/flink/configuration/ConfigurationParsingInvalidFormatsTest.java [hunk 4]
```diff
@@ -64,35 +66,35 @@ public static Object[][] getSpecs() {
         };
     }
 
-    @Parameterized.Parameter public ConfigOption<?> option;
+    @Parameter public ConfigOption<?> option;
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/configuration/ConfigurationParsingInvalidFormatsTest.java:69]: visibility can be private.

File: flink-core/src/test/java/org/apache/flink/configuration/FilesystemSchemeConfigTest.java [hunk 5]
```diff
@@ -21,83 +21,75 @@
...
-import static org.junit.Assert.fail;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
 /** Tests for the configuration of the default file system scheme. */
-public class FilesystemSchemeConfigTest extends TestLogger {
+public class FilesystemSchemeConfigTest {
 
-    @Rule public final TemporaryFolder tempFolder = new TemporaryFolder();
+    @TempDir public File tempFolder;
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/configuration/FilesystemSchemeConfigTest.java:38]: private

**Review by Jiabao-Sun — COMMENTED** [review]
Hi @GOODBOY008, 
Can we minimize the visibility of classes and methods?

**Review by 1996fanrui — APPROVED** [review]
Thanks @GOODBOY008 for the update!

LGTM.

File: flink-core/src/test/java/org/apache/flink/configuration/MemorySizePrettyPrintingTest.java [hunk 6]
```diff
@@ -45,13 +46,13 @@ public static Object[][] parameters() {
         };
     }
 
-    @Parameterized.Parameter public MemorySize memorySize;
+    @Parameter private MemorySize memorySize;
 
-    @Parameterized.Parameter(1)
+    @Parameter(value = 1)
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/configuration/MemorySizePrettyPrintingTest.java:51]: value can be ignored and expectedString can be private.

File: flink-core/src/test/java/org/apache/flink/configuration/ReadableWritableConfigurationTest.java [hunk 7]
```diff
@@ -169,7 +170,7 @@ void testGetOptionalFromObject() {
         testSpec.setValue(configuration);
 
         Optional<?> optional = configuration.getOptional(testSpec.getOption());
-        assertThat(optional.get(), equalTo(testSpec.getValue()));
+        assertThat(optional.get()).isEqualTo(testSpec.getValue());
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/configuration/ReadableWritableConfigurationTest.java:173]:
```suggestion
        assertThat(optional).hasValue(testSpec.getValue());
```

File: flink-core/src/test/java/org/apache/flink/configuration/GlobalConfigurationTest.java [hunk 8]
```diff
@@ -42,57 +42,48 @@ class GlobalConfigurationTest {
...
+        assertThat(conf.getString("mykey4", null)).isEqualTo("myvalue4");
+        assertThat(conf.getString("mykey5", null)).isEqualTo("myvalue5");
+        assertThat(conf.getString("mykey6", null)).isEqualTo("my: value6");
+        assertThat(conf.getString("mykey7", "null")).isEqualTo("null");
+        assertThat(conf.getString("mykey8", "null")).isEqualTo("null");
+        assertThat(conf.getString("mykey9", null)).isEqualTo("myvalue10");
+        // Clear the standard yaml flag to avoid impact to other cases.
+        GlobalConfiguration.setStandardYaml(true);
+        confFile.delete();
+        tmpDir.delete();
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/configuration/GlobalConfigurationTest.java:86]: In exceptional scenarios, will the deletion of temporary directories be ignored?

**Review by Jiabao-Sun — APPROVED** [review]
Thanks @GOODBOY008 for the great work and @1996fanrui for the review.
LGTM.

**Pull request merged**